main.tf

```diff
@@ -1 +1,43 @@
+locals {
+  registry_domain = length(var.registry_domain) > 0 ? var.registry_domain : (
+    "${data.aws_caller_identity.current.account_id}.dkr.ecr.${data.aws_region.current.name}.amazonaws.com"
+  )
+  iam_role_arn         = module.lacework_ecr_iam_role.created ? module.lacework_ecr_iam_role.arn : var.iam_role_arn
+  iam_role_name        = module.lacework_ecr_iam_role.created ? module.lacework_ecr_iam_role.name : var.iam_role_name
+  iam_role_external_id = module.lacework_ecr_iam_role.created ? module.lacework_ecr_iam_role.external_id : var.iam_role_external_id
+}
 
+data "aws_region" "current" {}
+data "aws_caller_identity" "current" {}
+
+module "lacework_ecr_iam_role" {
+  source                  = "lacework/iam-role/aws"
+  version                 = "~> 0.2"
+  create                  = var.use_existing_iam_role ? false : true
+  iam_role_name           = var.iam_role_name
+  lacework_aws_account_id = var.lacework_aws_account_id
+  external_id_length      = var.external_id_length
+  tags                    = var.tags
+}
+
+resource "aws_iam_role_policy_attachment" "ecr_read_only_policy_attachment" {
```

Comment:
Let's suppose an AWS account with multiple regions. Let's suppose we don't want to create one IAM role per region and that we want to reuse the same role in each region.

To achieve this goal, I would set `use_existing_iam_role` to `true`. It will work in the first region.

How will it reach in the second region?

I think this attachment will fail because the policy `AmazonEC2ContainerRegistryReadOnly` will already be attached.

Reply:
Very interesting use case! I will test it right away!

Reply:
It works like a charm!

Code:
```hcl
provider "lacework" {
  profile = "mini"
}

provider "aws" {
  alias  = "west2"
  region = "us-west-2"
}

module "lacework_ecr_west" {
  source = "../.."
  providers = {
    aws = aws.west2
  }
}

provider "aws" {
  alias  = "east1"
  region = "us-east-1"
}

module "lacework_ecr_east" {
  source = "../.."
  providers = {
    aws = aws.west2
  }
  use_existing_iam_role = true
  iam_role_name         = module.lacework_ecr_west.iam_role_name
  iam_role_arn          = module.lacework_ecr_west.iam_role_arn
  iam_role_external_id  = module.lacework_ecr_west.external_id
}
```
Running Terraform:
```
$ terraform apply
module.lacework_ecr_west.module.lacework_ecr_iam_role.random_id.uniq: Refreshing state... [id=uTkifw]
module.lacework_ecr_east.module.lacework_ecr_iam_role.random_id.uniq: Refreshing state... [id=mNsUIw]
module.lacework_ecr_west.module.lacework_ecr_iam_role.random_string.external_id[0]: Refreshing state... [id=VmfEQMbZ8Ws95Enq]
module.lacework_ecr_east.data.aws_caller_identity.current: Refreshing state...
module.lacework_ecr_east.data.aws_region.current: Refreshing state...
module.lacework_ecr_west.data.aws_region.current: Refreshing state...
module.lacework_ecr_west.data.aws_caller_identity.current: Refreshing state...
module.lacework_ecr_east.aws_iam_role_policy_attachment.ecr_read_only_policy_attachment: Refreshing state... [id=lw-iam-b939227f-20210316210253370700000001]
module.lacework_ecr_west.module.lacework_ecr_iam_role.data.aws_iam_policy_document.lacework_assume_role_policy[0]: Refreshing state...
module.lacework_ecr_west.module.lacework_ecr_iam_role.aws_iam_role.lacework_iam_role[0]: Refreshing state... [id=lw-iam-b939227f]
module.lacework_ecr_east.time_sleep.wait_time: Refreshing state... [id=2021-03-16T21:03:08Z]
module.lacework_ecr_west.aws_iam_role_policy_attachment.ecr_read_only_policy_attachment: Refreshing state... [id=lw-iam-b939227f-20210316210148989500000001]
module.lacework_ecr_west.lacework_integration_ecr.iam_role: Refreshing state... [id=MINIALLY_9293C0545824CF31E4D29AA3D2BE425189ED2D56F45B0B0]
module.lacework_ecr_east.lacework_integration_ecr.iam_role: Refreshing state... [id=MINIALLY_3884B70BF5F60803A1EB9A7B5238F78FFFD8DE746C2D6DC]
module.lacework_ecr_west.time_sleep.wait_time: Refreshing state... [id=2021-03-16T21:02:04Z]

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```
Two ECR integrations with the same IAM role in different Regions! ⭐
```
$ lacework int -p mini show MINIALLY_9293C0545824CF31E4D29AA3D2BE425189ED2D56F45B0B0
                      INTEGRATION GUID                            NAME             TYPE        STATUS    STATE
-----------------------------------------------------------+-----------------+---------------+---------+--------
  MINIALLY_9293C0545824CF31E4D29AA3D2BE425189ED2D56F45B0B0   TF ECR IAM Role   CONT_VULN_CFG   Enabled   Ok

                               INTEGRATION DETAILS
----------------------------------------------------------------------------------
    REGISTRY TYPE               AWS_ECR
    REGISTRY DOMAIN             463783698038.dkr.ecr.us-west-2.amazonaws.com
    LIMIT BY TAG                *
    LIMIT BY LABEL              *
    LIMIT NUM IMAGES PER REPO   5
    AWS AUTH TYPE               AWS_IAM
    ROLE ARN                    arn:aws:iam::463783698038:role/lw-iam-b939227f
$
$ lacework int -p mini show MINIALLY_3884B70BF5F60803A1EB9A7B5238F78FFFD8DE746C2D6DC
                      INTEGRATION GUID                            NAME             TYPE        STATUS    STATE
-----------------------------------------------------------+-----------------+---------------+---------+--------
  MINIALLY_3884B70BF5F60803A1EB9A7B5238F78FFFD8DE746C2D6DC   TF ECR IAM Role   CONT_VULN_CFG   Enabled   Ok

                               INTEGRATION DETAILS
----------------------------------------------------------------------------------
    REGISTRY TYPE               AWS_ECR
    REGISTRY DOMAIN             463783698038.dkr.ecr.us-west-2.amazonaws.com
    LIMIT BY TAG                *
    LIMIT BY LABEL              *
    LIMIT NUM IMAGES PER REPO   5
    AWS AUTH TYPE               AWS_IAM
    ROLE ARN                    arn:aws:iam::463783698038:role/lw-iam-b939227f
```
@JPLachance Thoughts?

Reply:
I will add this as an example!

![tenor-114225924](https://user-images.githubusercontent.com/5712253/111380315-c9c8cc00-8669-11eb-8356-0b59e2c4bc40.gif)

Reply:
LGTM!

Reply:
Great stuff!